src/main/scala/org/broadinstitute/dsde/firecloud/trial/ProjectManager.scala

```diff
@@ -0,0 +1,153 @@
+      logger.info(s"project creation cycle: $getCurrentStatus")
+
+      // generate a unique project name
+      val uniqueProjectName = getAndRecordUniqueProjectName
+
+      uniqueProjectName map { projectName =>
+        logger.info(s"creating name <$projectName> via rawls ...")
+        // create project via rawls, after sudoing to the trial billing manager
+        val saToken:WithAccessToken = AccessToken(OAuth2BearerToken(HttpGoogleServicesDAO.getTrialBillingManagerAccessToken))
+        rawlsDAO.createProject(projectName, billingAcct)(saToken) map { createSuccess =>
+          if (createSuccess) {
+            logger.info(s"rawls acknowledged create request for <$projectName>.")
+            // schedule a verify for the project we just created
+            context.system.scheduler.scheduleOnce(verifyDelay, self, Verify(projectName))
+          } else {
+            logger.warn(s"rawls reports error when creating <$projectName>.")
+          }
+        }
+        // schedule the next creation
+        context.system.scheduler.scheduleOnce(createDelay, self, Create(current+1, target))
+      }
+    }
+  }
+
+  private def verify(projectName: String) = {
+    logger.info(s"verifying project <$projectName> ...")
+    val rawlsName = RawlsBillingProjectName(projectName)
+    val saToken:WithAccessToken = AccessToken(OAuth2BearerToken(HttpGoogleServicesDAO.getTrialBillingManagerAccessToken))
+    // as the trial billing manager, get the list of all projects I own (no API to get a single project)
+    rawlsDAO.getProjects(saToken) map { projects =>
+      projects.find(p => p.projectName == rawlsName) match {
+        case None => // project doesn't exist in rawls. This shouldn't happen.
+          logger.warn(s"Project <$projectName> was created in rawls but not returned by rawls. Calling it an error.")
+          trialDAO.setProjectRecordVerified(rawlsName, verified = true, status = CreationStatuses.Error)
+        case Some(proj) =>
+          proj.creationStatus match {
+            case CreationStatuses.Creating =>
+              logger.info(s"Project <$projectName> still creating; will retry verification.")
+              context.system.scheduler.scheduleOnce(verifyDelay, self, Verify(projectName))
+            case CreationStatuses.Ready =>
+              logger.info(s"Project <$projectName> verified as ready!")
+              trialDAO.setProjectRecordVerified(rawlsName, verified = true, status = CreationStatuses.Ready)
+            case CreationStatuses.Error =>
+              logger.warn(s"Project <$projectName> errored during creation: ${proj.message}")
+              trialDAO.setProjectRecordVerified(rawlsName, verified = true, status = CreationStatuses.Error)
+          }
+      }
+    } recover {
+      case t:Throwable =>
+        logger.warn(s"Error checking status on <$projectName>: ${t.getMessage}")
+        context.system.scheduler.scheduleOnce(verifyDelay, self, Verify(projectName))
```

Comment:
should we only retry this a few times?

Reply:
good call, I added a fairly lenient limit on retries for this.